### PR TITLE
NIFI-3392 Enhance documentation for provenance event type definitions

### PR DESCRIPTION
**Pull request by andrewmlim**
Added new "Provenance Events" sections to User and Dev Guides.  Made a minor edit in Admin guide for consistent use of "Provenance events".

### CODE REVIEW

**Comment by andrewmlim** [comment]
When reviewing, please note that the tables added to the User and Dev Guides, while similar, are actually different.  The User Guide has more simple descriptions of the provenance event types, whereas, the Dev Guide has more dev relevant material included from https://github.com/apache/nifi/blob/master/nifi-api/src/main/java/org/apache/nifi/provenance/ProvenanceEventType.java

File: nifi-docs/src/main/asciidoc/administration-guide.adoc [hunk 1]
```diff
@@ -2074,7 +2074,7 @@ The Provenance Repository contains the information related to Data Provenance. T
 
 |====
 |*Property*|*Description*
-|nifi.provenance.repository.implementation|The Provenance Repository implementation. The default value is org.apache.nifi.provenance.PersistentProvenanceRepository and should only be changed with caution. To store provenance events in memory instead of on disk (at the risk of data loss in the event of power/machine failure), set this property to org.apache.nifi.provenance.VolatileProvenanceRepository.
+|nifi.provenance.repository.implementation|The Provenance Repository implementation. The default value is org.apache.nifi.provenance.PersistentProvenanceRepository and should only be changed with caution. To store Provenance events in memory instead of on disk (at the risk of data loss in the event of power/machine failure), set this property to org.apache.nifi.provenance.VolatileProvenanceRepository.
```
Comment (alopresto) [inline nifi-docs/src/main/asciidoc/administration-guide.adoc]: Not sure why the capitalization changed here but I'm reverting; "provenance" is not a proper noun.

File: nifi-docs/src/main/asciidoc/user-guide.adoc [hunk 2]
```diff
@@ -1690,11 +1687,36 @@ replay data at any point within the dataflow, and see a graphical representation
...
+|CONTENT_MODIFIED        |Indicates that a FlowFile's content was modified in some way
+|CREATE                  |Indicates that a FlowFile was generated from data that was not received from a remote system or external process
+|DOWNLOAD                |Indicates that the contents of a FlowFile were downloaded by a user or external entity
+|DROP                    |Indicates a provenance event for the conclusion of an object's life for some reason other than object expiration
+|EXPIRE                  |Indicates a provenance event for the conclusion of an object's life due to the object not being processed in a timely manner
+|FETCH                   |Indicates that the contents of a FlowFile were overwritten using the contents of some external resource
+|FORK                    |Indicates that one or more FlowFiles were derived from a parent FlowFile
+|JOIN                    |Indicates that a single FlowFile is derived from joining together multiple parent FlowFiles
+|RECEIVE                 |Indicates a provenance event for receiving data from an external process
+|REPLAY                  |Indicates a provenance event for replaying a FlowFile.
```
Comment (alopresto) [inline nifi-docs/src/main/asciidoc/user-guide.adoc]: Removing unnecessary '.'.

File: nifi-docs/src/main/asciidoc/developer-guide.adoc [hunk 3]
```diff
@@ -848,6 +848,31 @@ Because this documentation is in an HTML format, you may include images and tabl
...
 documentation for Processors, ControllerServices and ReportingTasks.
 
+[[provenance_events]]
+== Provenance Events
+
+The different event types for provenance reporting are:
+
+[options="header"]
+|======================
+|Provenance Event        |Description
```
Comment (alopresto) [inline nifi-docs/src/main/asciidoc/developer-guide.adoc]: Making line endings consistent (removing all trailing '.').

**Pull request closed without merging**